src/main.rs

```diff
@@ -188,14 +250,20 @@ fn build_sysroot(auto: bool, target: &str, rustc_version: &VersionMeta) -> PathB
 
     // Determine where to put the sysroot.
     let user_dirs = directories::ProjectDirs::from("de", "ralfj", "cargo-careful").unwrap();
-    let sysroot_dir = user_dirs.cache_dir();
+    let mut sysroot_dir: PathBuf = user_dirs.cache_dir().to_owned();
 
-    // Do the build.
-    eprint!("Preparing a careful sysroot (target: {target})... ");
+    if let Some(san) = sanitizer {
+        sysroot_dir.push(san);
+        eprint!("Preparing a careful sysroot (target: {target}, sanitizer: {san})... ")
+    } else {
+        eprint!("Preparing a careful sysroot (target: {target})... ")
+    }
     if !auto {
         eprintln!();
     }
-    Sysroot::new(sysroot_dir, target)
+
+    // Do the build.
```

Comment:
I am fine with making this a separate 'paragraph' but then please also add an empty line after the `build_from_source` and before the subsequent printing.